web/web-mock/web-mock-rest/src/main/java/com/castlemock/web/mock/rest/controller/mock/AbstractRestServiceController.java

```diff
@@ -262,45 +242,11 @@ private RestResponse forwardRequest(final RestRequest request,
         }
 
 
-
-        HttpURLConnection connection = null;
-        try {
-
-            String requestBody = null;
-
-            if (HttpMethod.POST.equals(request.getHttpMethod()) ||
-                    HttpMethod.PUT.equals(request.getHttpMethod()) ||
-                    HttpMethod.DELETE.equals(request.getHttpMethod())) {
-                requestBody = request.getBody();
-            }
-
-            final String parameterUri = HttpMessageSupport.buildParameterUri(request.getHttpParameters());
-            final String endpoint = restMethod.getForwardedEndpoint() + request.getUri() + parameterUri;
-
-            connection = HttpMessageSupport.establishConnection(
-                    endpoint,
-                    request.getHttpMethod(),
-                    requestBody,
-                    request.getHttpHeaders());
-
-            final List<ContentEncoding> encodings = HttpMessageSupport.extractContentEncoding(connection);
-            final List<HttpHeader> responseHttpHeaders = HttpMessageSupport.extractHttpHeaders(connection);
-            final String characterEncoding = CharsetUtility.parseHttpHeaders(responseHttpHeaders);
-            final String responseBody = HttpMessageSupport.extractHttpBody(connection, encodings, characterEncoding);
-            return RestResponse.builder()
-                    .body(responseBody)
-                    .mockResponseName(FORWARDED_RESPONSE_NAME)
-                    .httpHeaders(responseHttpHeaders)
-                    .httpStatusCode(connection.getResponseCode())
-                    .contentEncodings(encodings)
-                    .build();
-        } catch (IOException exception) {
-            LOGGER.error("Unable to forward request", exception);
+        Optional<RestResponse> optionalRestResponse = restClient.getResponse(request, restMethod);
```

Comment:
Could make this a little bit shorter by doing something like this>
`return restClient.getResponse(request, restMethod).orElseThrow(() -> new RestException("Unable to forward request to configured endpoint");)`